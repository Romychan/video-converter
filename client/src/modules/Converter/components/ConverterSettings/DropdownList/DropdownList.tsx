import {
  VIDEO_FORMAT,
  VIDEO_QUALITY,
  VIDEO_FPS,
  VIDEO_SOUND,
} from '../../../constants/video';

import { useActions } from '@hooks/useActions';

import { Accordion } from '@ui/Accordion/Accordion';
import { Dropdown } from '@ui/Dropdown/Dropdown';
import { DropdownItem } from '@ui/Dropdown/DropdownItem';

import styles from './DropdownList.module.scss';

export const DropdownList = () => {
  const { changeFormat, changeFps, changeQuality, changeAudio } = useActions();

  return (
    <>
      <Dropdown label="Формат" onChange={(item) => changeFormat(item)}>
        {VIDEO_FORMAT.map((item) => (
          <DropdownItem key={item.value} value={item.value}>
            {item.label}
          </DropdownItem>
        ))}
      </Dropdown>

      <Dropdown label="Качество" onChange={(item) => changeQuality(item)}>
        {VIDEO_QUALITY.map((item) => (
          <DropdownItem key={item.value} value={item.value}>
            {item.label}
          </DropdownItem>
        ))}
      </Dropdown>

      <Accordion title="Дополнительные настройки">
        <div className={styles.container}>
          <Dropdown
            label="Кадры в секунду (FPS)"
            onChange={(item) => changeFps(item)}
          >
            {VIDEO_FPS.map((item) => (
              <DropdownItem key={item.value} value={item.value}>
                {item.label}
              </DropdownItem>
            ))}
          </Dropdown>

          <Dropdown label="Аудиоканал" onChange={(item) => changeAudio(item)}>
            {VIDEO_SOUND.map((item) => (
              <DropdownItem key={item.value} value={item.value}>
                {item.label}
              </DropdownItem>
            ))}
          </Dropdown>
        </div>
      </Accordion>
    </>
  );
};
