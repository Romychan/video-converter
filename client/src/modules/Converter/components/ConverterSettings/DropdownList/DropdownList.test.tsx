import { RenderResult, fireEvent, screen } from '@testing-library/react';

import { renderWithProviders } from '@tests/helpers/renderWithProviders';

import { DropdownList } from './DropdownList';

describe('DropdownList', () => {
  let container: RenderResult;

  beforeEach(() => {
    container = renderWithProviders(<DropdownList />);
  });

  it('should change value in dropdown', async () => {
    const qualityDropdownElement = screen.getByText(/360/i);

    fireEvent.click(qualityDropdownElement);
    fireEvent.click(screen.getByText(/720/i));

    expect(screen.getByText(/720/i)).toBeInTheDocument();
  });

  it('should change value in accordion', async () => {
    const accordionElement = screen.getByText(/Дополнительные настройки/i);
    fireEvent.click(accordionElement);

    const fpsDropdownElement = screen.getByText(/24 FPS/i);
    fireEvent.click(fpsDropdownElement);

    fireEvent.click(screen.getByText(/60 FPS/i));

    expect(screen.getByText(/60 FPS/i)).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
