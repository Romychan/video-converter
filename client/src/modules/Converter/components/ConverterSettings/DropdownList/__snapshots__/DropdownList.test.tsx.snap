// Vitest Snapshot v1

exports[`DropdownList > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h3
          class="_label_b2c890"
          data-testid="dropdown-label"
        >
          Формат
        </h3>
        <div
          class="_dropdown_b2c890"
          data-testid="dropdown-container"
        >
          <div
            class="_title_b2c890"
            data-testid="dropdown-header"
          >
            .MP4
            <svg
              class="icon icon-arrow _icon_b2c890"
              data-testid="icon"
              height="11"
              stroke=""
              width="11"
            >
              <use
                xlink:href="/src/assets/img/general/icons.svg#arrow"
              />
            </svg>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="_label_b2c890"
          data-testid="dropdown-label"
        >
          Качество
        </h3>
        <div
          class="_dropdown_b2c890"
          data-testid="dropdown-container"
        >
          <div
            class="_title_b2c890"
            data-testid="dropdown-header"
          >
            360p
            <svg
              class="icon icon-arrow _icon_b2c890"
              data-testid="icon"
              height="11"
              stroke=""
              width="11"
            >
              <use
                xlink:href="/src/assets/img/general/icons.svg#arrow"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="_accordion_7e54ec"
        data-testid="accordion"
      >
        <div
          class="_header_7e54ec"
          data-testid="accordion-header"
        >
          <div
            class="_square_7e54ec"
          >
            <svg
              class="icon icon-plus _icon_7e54ec"
              data-testid="icon"
              height="14"
              stroke=""
              width="14"
            >
              <use
                xlink:href="/src/assets/img/general/icons.svg#plus"
              />
            </svg>
          </div>
          <h3
            class="_title_7e54ec"
          >
            Дополнительные настройки
          </h3>
        </div>
        <div
          class="_body_7e54ec"
          data-testid="accordion-body"
          style="max-height: 0px;"
        >
          <div
            class="_content_7e54ec"
          >
            <div
              class="_container_02ec06"
            >
              <div>
                <h3
                  class="_label_b2c890"
                  data-testid="dropdown-label"
                >
                  Кадры в секунду (FPS)
                </h3>
                <div
                  class="_dropdown_b2c890"
                  data-testid="dropdown-container"
                >
                  <div
                    class="_title_b2c890"
                    data-testid="dropdown-header"
                  >
                    24 FPS
                    <svg
                      class="icon icon-arrow _icon_b2c890"
                      data-testid="icon"
                      height="11"
                      stroke=""
                      width="11"
                    >
                      <use
                        xlink:href="/src/assets/img/general/icons.svg#arrow"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div>
                <h3
                  class="_label_b2c890"
                  data-testid="dropdown-label"
                >
                  Аудиоканал
                </h3>
                <div
                  class="_dropdown_b2c890"
                  data-testid="dropdown-container"
                >
                  <div
                    class="_title_b2c890"
                    data-testid="dropdown-header"
                  >
                    Стерео
                    <svg
                      class="icon icon-arrow _icon_b2c890"
                      data-testid="icon"
                      height="11"
                      stroke=""
                      width="11"
                    >
                      <use
                        xlink:href="/src/assets/img/general/icons.svg#arrow"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h3
        class="_label_b2c890"
        data-testid="dropdown-label"
      >
        Формат
      </h3>
      <div
        class="_dropdown_b2c890"
        data-testid="dropdown-container"
      >
        <div
          class="_title_b2c890"
          data-testid="dropdown-header"
        >
          .MP4
          <svg
            class="icon icon-arrow _icon_b2c890"
            data-testid="icon"
            height="11"
            stroke=""
            width="11"
          >
            <use
              xlink:href="/src/assets/img/general/icons.svg#arrow"
            />
          </svg>
        </div>
      </div>
    </div>
    <div>
      <h3
        class="_label_b2c890"
        data-testid="dropdown-label"
      >
        Качество
      </h3>
      <div
        class="_dropdown_b2c890"
        data-testid="dropdown-container"
      >
        <div
          class="_title_b2c890"
          data-testid="dropdown-header"
        >
          360p
          <svg
            class="icon icon-arrow _icon_b2c890"
            data-testid="icon"
            height="11"
            stroke=""
            width="11"
          >
            <use
              xlink:href="/src/assets/img/general/icons.svg#arrow"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="_accordion_7e54ec"
      data-testid="accordion"
    >
      <div
        class="_header_7e54ec"
        data-testid="accordion-header"
      >
        <div
          class="_square_7e54ec"
        >
          <svg
            class="icon icon-plus _icon_7e54ec"
            data-testid="icon"
            height="14"
            stroke=""
            width="14"
          >
            <use
              xlink:href="/src/assets/img/general/icons.svg#plus"
            />
          </svg>
        </div>
        <h3
          class="_title_7e54ec"
        >
          Дополнительные настройки
        </h3>
      </div>
      <div
        class="_body_7e54ec"
        data-testid="accordion-body"
        style="max-height: 0px;"
      >
        <div
          class="_content_7e54ec"
        >
          <div
            class="_container_02ec06"
          >
            <div>
              <h3
                class="_label_b2c890"
                data-testid="dropdown-label"
              >
                Кадры в секунду (FPS)
              </h3>
              <div
                class="_dropdown_b2c890"
                data-testid="dropdown-container"
              >
                <div
                  class="_title_b2c890"
                  data-testid="dropdown-header"
                >
                  24 FPS
                  <svg
                    class="icon icon-arrow _icon_b2c890"
                    data-testid="icon"
                    height="11"
                    stroke=""
                    width="11"
                  >
                    <use
                      xlink:href="/src/assets/img/general/icons.svg#arrow"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div>
              <h3
                class="_label_b2c890"
                data-testid="dropdown-label"
              >
                Аудиоканал
              </h3>
              <div
                class="_dropdown_b2c890"
                data-testid="dropdown-container"
              >
                <div
                  class="_title_b2c890"
                  data-testid="dropdown-header"
                >
                  Стерео
                  <svg
                    class="icon icon-arrow _icon_b2c890"
                    data-testid="icon"
                    height="11"
                    stroke=""
                    width="11"
                  >
                    <use
                      xlink:href="/src/assets/img/general/icons.svg#arrow"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
